src/lib/useDragAndZoom.tsx

```diff
@@ -119,7 +126,13 @@ export function useDragAndZoom(
                     const k = (prev1 - prev2) / (new1 - new2);
                     const b = prev1 - k * new1;
 
-                    temporaryViewport = [k * temporaryViewport[0] + b, k * temporaryViewport[1] + b];
+                    const newViewport: Bounds = [k * temporaryViewport[0] + b, k * temporaryViewport[1] + b]
+
+                    if (!checkBoundsInMinVisibleX(newViewport, options.boundsMinVisibleX)) {
+                        return;
+                    }
```

Comment:
:pinching_hand: It is not so important, but I suggest to adjust `newViewport` to match `minVisibleX` requirement instead of just cancelling the zooming. Something like this:

```suggestion
                    if (!checkBoundsInMinVisibleX(newViewport, options.boundsMinVisibleX)) {
                        const center = (newViewport[1] - newViewport[0]) / 2;
                        newViewport[0] = center - options.boundsMinVisibleX / 2;
                        newViewport[1] = center + options.boundsMinVisibleX / 2;
                    }
```

Without this fast zooming will stop at arbitrary zoom level, not necessary at configured `minVisibleX`.

Maybe it worth converting `checkBoundsInMinVisibleX()` that only checks the condition to `fitBoundsToMinVisibleX()` that will take the new viewport bounds and adapt it if necessary (like `fitBoundsInLimit()`).